tests/lib/rules/require-default-props.js

```diff
@@ -1306,7 +1306,26 @@ ruleTester.run('require-default-props', rule, {
         column: 3
       }]
     },
-
+    //
+    // component with no pros followed by a failing component
+    {
+      code: [
+        'var ComponentWithNoProps = ({ bar = "bar" }) => {',
```

Comment:
It reads a bit confusing. It says`component with no props` but this component actually has props: `bar` and `foo`. I guess that you actually mean; `component with no declared prop types` (since there is no `.propTypes`) for the first component.